Civi/ActionSchedule/RecipientBuilder.php

```diff
@@ -581,6 +581,16 @@ protected function selectActionLogFields($phase, $query) {
             "#casActionScheduleId as action_schedule_id",
           )
         );
+        //CRM-19017: Load default params for fragment query object.
+        $params = array(
+          'casActionScheduleId' => $this->actionSchedule->id,
+          'casNow' => $this->now,
+        );
+        foreach ($params as $key => $val) {
+          if (empty($fragment->params[$key])) {
+            $fragment->param($key, $val);
+          }
+        }
```

Comment:
For point 2 and 3 - by loading `casActionScheduleId` and `casNow` in the fragment query object.